sentry/src/main/java/io/sentry/instrumentation/file/SentryFileInputStream.java

```diff
@@ -0,0 +1,197 @@
+package io.sentry.instrumentation.file;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.ISpan;
+import io.sentry.Sentry;
+import io.sentry.SpanStatus;
+import io.sentry.util.StringUtils;
+import java.io.File;
+import java.io.FileDescriptor;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * An implementation of {@link java.io.FileInputStream} that creates a {@link io.sentry.ISpan} for
+ * reading operation with filename and byte count set as description
+ *
+ * Note, that span is started when this InputStream is instantiated via constructor and finishes
+ * when the {@link java.io.FileInputStream#close()} is called.
+ */
+@SuppressWarnings("unused") // used via bytecode manipulation (SAGP)
+@Open
+public class SentryFileInputStream extends FileInputStream {
+
+  private final @Nullable ISpan currentSpan;
+  private final @Nullable File file;
+  private final @NotNull FileInputStream delegate;
+
+  private @NotNull SpanStatus spanStatus = SpanStatus.OK;
```

Comment:
Not sure if we should handle concurrency here - generally, it's discouraged to use IOStreams across multiple threads

Reply:
good question, lets discuss with @bruno-garcia too since he always has opinions about that

Reply:
`FileStreamInput` can't/isn't designed be used concurrently (not without synchronization) so I think we're safe here.